src/main/python/covata/delta/client.py

```diff
@@ -0,0 +1,144 @@
+#   Copyright 2017 Covata Limited or its affiliates
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+
+from __future__ import absolute_import
+
+from . import crypto
+from . import utils
+
+
+class Client(utils.LogMixin):
+    """
+    The main entry point for the Delta SDK.
+
+    An instance of this class will provide an interface to work and interact
+    with the Delta API. The core domain objects (Identity, Secret and
+    Event) are returned from method calls to this class, and themselves provide
+    fluent interface that can be used to continue interactive with the Delta
+    API. Consumers of this SDK can therefore choose whether they wish to
+    construct all the calls from base values (i.e. id strings such as
+    identity_id, secret_id, etc) or via the fluent interfaces (or a mixture of
+    both).
+    """
+
+    def __init__(self, config):
+        """
+        Creates a new DeltaClient instance from the provided configuration.
+
+        :param config: the configuration for the client
+        :type config: dict[str, any]
+        """
+        self.__key_store = config["key_store"]
+        self.__api_client = config["api_client"]
+
+    @property
+    def key_store(self):
+        return self.__key_store
+
+    @property
+    def api_client(self):
+        return self.__api_client
+
+    def create_identity(self, external_id=None, metadata=None):
+        """
+        Creates a new identity in Delta.
+
+        :param external_id: the external id to associate with the identity
+        :param metadata: the metadata to associate with the identity
+        :type external_id: str | None
+        :type metadata: dict[str, str] | None
+        :return: the identity
+        :rtype: :class:`Identity`
```

Comment:
need to be modified to below to ensure sphinx put the proper links in the docs.

```
:rtype: :class:`~.Identity`
```

the tilde `~.` indicates the current module